kernel/src/hil/time.rs

```diff
@@ -714,6 +714,76 @@ impl PartialEq for Ticks64 {
 
 impl Eq for Ticks64 {}
 
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum DayOfWeek {
+    Sunday,
+    Monday,
+    Tuesday,
+    Wednesday,
+    Thursday,
+    Friday,
+    Saturday,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum Month {
+    January,
+    February,
+    March,
+    April,
+    May,
+    June,
+    July,
+    August,
+    September,
+    October,
+    November,
+    December,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct DateTime {
+    pub year: u16,
+    pub month: Month,
+    pub day: u8,
+    pub day_of_week: DayOfWeek,
+    pub hour: u8,
+    pub minute: u8,
+    pub seconds: u8,
+}
+
+/// Interface for reading and setting the current time
+pub trait Rtc<'a> {
```

Comment:
Modifying the time HIL is complicated, and we probably want to avoid that. I think it would be better to just create a new real time clock HIL file and put this there.